From ae298acdeadbb2e38b5f1eff62762cf2a3b1baab Mon Sep 17 00:00:00 2001
From: Fei Jiang <fei.jiang@intel.com>
Date: Mon, 13 Aug 2018 14:24:23 +0800
Subject: [PATCH 2019/2313] drm/i915/gvt: fix display messy issue which is
 introduced by plane pvmmio

For plane pvmmio optimization, we need cache all plane related registers,
in previous commit 9c3e8b1a3f15 ("drm/i915/gvt: handling pvmmio update of
plane registers in GVT-g"), PLANE_AUX_DIST and PLANE_AUX_OFFSET are missing
to be handled.

Signed-off-by: Fei Jiang <fei.jiang@intel.com>
Reviewed-by: He, Min <min.he@intel.com>
Reviewed-by: Zhao Yakui <yakui.zhao@intel.com>
---
 drivers/gpu/drm/i915/gvt/handlers.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/drivers/gpu/drm/i915/gvt/handlers.c b/drivers/gpu/drm/i915/gvt/handlers.c
index c33c771..53ae0c6c 100644
--- a/drivers/gpu/drm/i915/gvt/handlers.c
+++ b/drivers/gpu/drm/i915/gvt/handlers.c
@@ -777,6 +777,12 @@ static void pvmmio_update_plane_register(struct intel_vgpu *vgpu,
 	skl_plane_mmio_write(vgpu,
 		i915_mmio_reg_offset(PLANE_SIZE(pipe, plane)),
 		&pv_plane->plane_size, 4);
+	skl_plane_mmio_write(vgpu,
+		i915_mmio_reg_offset(PLANE_AUX_DIST(pipe, plane)),
+		&pv_plane->plane_aux_dist, 4);
+	skl_plane_mmio_write(vgpu,
+		i915_mmio_reg_offset(PLANE_AUX_OFFSET(pipe, plane)),
+		&pv_plane->plane_aux_offset, 4);
 
 	if (pv_plane->flags & PLANE_SCALER_BIT) {
 		skl_ps_mmio_write(vgpu,
-- 
2.7.4

